src/balances/sui.ts

```diff
@@ -12,15 +12,55 @@ export async function pullSuiNativeBalance(conn: Connection, address: string): P
   } as WalletBalance;
 }
 
+// taken from: https://github.com/MystenLabs/sui/blob/818406c5abdf7de1b80915a0519071eec3a5b1c7/crates/sui-types/src/crypto.rs#L1650
+// see: https://github.com/MystenLabs/sui/blob/d1d6eba/sdk/typescript/src/cryptography/ed25519-keypair.ts#L65
+const keyPairsByHexPrefix = {
+  '0x00': buildEd25519KeyPair,
+  '0x01': buildSecp256k1KeyPair,
+}
+
 export function pullSuiTokenBalance() {
   throw new Error('pullSuiTokenBalance is not yet implemented for SUI wallet');
 }
 
+function buildSecp256k1KeyPair(key: Buffer): Secp256k1Keypair {
+  return Secp256k1Keypair.fromSecretKey(key);
+}
+
+function buildEd25519KeyPair(key: Buffer): Ed25519Keypair {
+  return Ed25519Keypair.fromSecretKey(key);
+}
+
 export function getSuiAddressFromPrivateKey(privateKey: string) {
-  let keyPair;
+  let parsedKey = Buffer.from(privateKey, 'base64');
+
+  let key, buildKeyPair;
+  if (parsedKey.length === PRIVATE_KEY_SIZE+1) {
+    key = parsedKey.slice(1);
+    const schemaPrefix =`0x${parsedKey.slice(0, 1).toString('hex')}`
+
+    buildKeyPair = keyPairsByHexPrefix[
+      schemaPrefix as keyof typeof keyPairsByHexPrefix
+    ];
 
+    if (!buildKeyPair) {
+      throw new Error(`Invalid Keypair: prefix ${schemaPrefix}`);
+    }
+  }
+
+  else if (parsedKey.length === PRIVATE_KEY_SIZE) {
+    key = parsedKey;
+    buildKeyPair = buildEd25519KeyPair;
+  }
+
+  else {
+    throw new Error(`Invalid Sui private key. Expected length: 32 or 33 bytes, actual length: ${parsedKey.length}`);
+  }
+
+
+  let keyPair;
   try {
-    keyPair = Secp256k1Keypair.fromSecretKey(Buffer.from(privateKey, 'base64'));
+    keyPair = buildKeyPair(key);
   } catch (error) {
     throw new Error(`Invalid Sui private key. Error: ${error}`);
   }
```

Comment:
Do we actually want to wrap this error? :thinking:
Note that this won't even preserve the stack trace of the original error.

Reply:
If we do wrap it I'd do it like this:

```suggestion
  try {
    keyPair = Secp256k1Keypair.fromSecretKey(Buffer.from(privateKey, 'base64'));
    keyPair = buildKeyPair(key);
  } catch (error) {
    throw new Error(`Invalid Sui private key. Error: ${(error as Error)?.stack || error}`);
  }
```

and perhaps even call `util.inspect` on the `error` when the stack trace is not available.